sled-agent/src/common/instance.rs

```diff
@@ -214,30 +174,20 @@ pub enum Action {
 
 impl InstanceStates {
     pub fn new(
-        instance: InstanceRuntimeState,
         vmm: VmmRuntimeState,
         propolis_id: PropolisUuid,
+        migration_id: Option<Uuid>,
```

Comment:
Is it necessary to supply this here? Waiting until the instances ensure request comes in would remove an awkward `expect` at ensure time, and it seems like this shouldn't be load-bearing (no one will publish any instance state changes to Nexus until the ensure request arrives anyway, and even if they did, they just wouldn't contain any migration status, which seems correct--there's nothing to update until the Propolis is running).

I get the feeling I'm overlooking something, though.

Reply:
@gjcolombo I had to page this back in for a moment, but I believe this is necessary because the migration ID *isn't* sent in the instance-ensure-state request (`PUT /instances/{id}/state`), it's sent in the *instance-register* request (`PUT /instances/{id}`). The ensure-state request has the source Propolis UUID and IP, but *not* the migration ID, and we *need* to grab the migration ID here so that we can send it to Propolis when we actually send the instance-ensure to Propolis.

I would be happy to go back and change this up so that the ensure-state request to sled-agent contains the migration ID, instead. At the time, I didn't want to substantially change the structure of those sled-agent API requests as part of this change --- but we certainly can. Alternatively, it could be done in a follow-up. What do you think?

Reply:
Ah, I see now. The question I was trying to ask (but asked badly) was "why not just wait for Propolis to report the migration in for the first time, like it does for migrations out?" And the answer is "`propolis_client::types::InstanceMigrateInitiateRequest` needs the migration ID, and that's not in the `InstanceMigrationTargetParams` that got passed to sled agent, so it has to come from the `InstanceStates` instead."

I don't think this is a big enough issue to worry about in this PR, but I'd at least file a tracking issue and refer to it near the `expect` I thought we could remove (in `propolis_ensure_inner`, sled-agent/src/instance.rs:649 in the latest changes).

Reply:
Okay, I've opened #6073 for this, and added comment referencing it in ee5a380b2402ef423793050fc21bc64f5357b113.